src/xenia/gpu/trace_player.h

```diff
@@ -34,8 +35,14 @@ class TracePlayer : public TraceReader {
   GraphicsSystem* graphics_system() const { return graphics_system_; }
   int current_frame_index() const { return current_frame_index_; }
   int current_command_index() const { return current_command_index_; }
+  bool playing_trace() const { return playing_trace_; }
   const Frame* current_frame() const;
 
+  // Only valid if playing_trace is true.
+  const uint8_t* player_current_ptr() const { return player_current_ptr_; }
```

Comment:
you could expose this as a playback_percent() instead that pulls from a member that is updated during playback. that way less of the internal state is exposed in the API (which has threading implications) and the risk of the data being out of date is low (no one can use the percent as a pointer, so if it's slightly off that's not terrible).